internal/operators/odf/odf_operator.go

```diff
@@ -129,14 +124,30 @@ func (o *operator) getValidDiskCount(disks []*models.Disk, installationDiskID st
 
 // ValidateHost verifies whether this operator is valid for given host
 func (o *operator) ValidateHost(_ context.Context, cluster *common.Cluster, host *models.Host, additionalOperatorRequirements *models.ClusterHostRequirementsDetails) (api.ValidationResult, error) {
-	// temporary disabling ODF for non-standad HA OCP Control Plane until it will be clear how ODF will work in this scenario.
-	// We pass host validation to avoid false validation messages. The cluster validation will fail
-	masters, _, _ := common.GetHostsByEachRole(&cluster.Cluster, true)
-	if masterCount := len(masters); masterCount > 3 {
+	mode := getODFDeploymentMode(&cluster.Cluster, o.config.ODFNumMinimumHosts)
+	hostEffectiveRole := common.GetEffectiveRole(host)
+	shouldHostRunODF, err := shouldHostRunODF(&cluster.Cluster, mode, hostEffectiveRole)
+
+	// host role is auto-assign in standard mode.
+	if err != nil {
+		status := "For ODF with more than 3 hosts, role must be assigned to master or worker."
```

Comment:
We are assuming internals of the function `shouldHostRunODF`. Maybe we can just print error message as status? `status := err.Error()`

Reply:
I have an issue with the error message can't start with capital but the status should. I don't want to add extra code func to capitalize the first letter

Reply:
I think we should just capitalize programmatically the first letter. This will make the code less fragile (more flexible to change) and less confusing (we won't ask ourselves "how come we know it's more than 3 hosts at this point?")